### PR TITLE
Add PDM to the list of alternatives to Pipenv

### PR DESCRIPTION
**Pull request by pawamoy**
We could also add https://github.com/FFY00/trampolim, but I wanted to see what you think first (and if we add more, all descriptions should probably be made either as comparisons to pipenv, or as pipenv-unrelated descriptions).
I believe these alternatives deserve an entry on this page :slightly_smiling_face:

### CODE REVIEW

**Review by henryiii — APPROVED** [review]

**Comment by pawamoy** [comment]
Ah, yes, seeing dependencies in their README example confused me. Hatch is already there ;)

File: source/tutorials/managing-dependencies.rst [hunk 1]
```diff
@@ -161,20 +159,26 @@ Other Tools for Application Dependency Management
...
 * `micropipenv <https://github.com/thoth-station/micropipenv>`_ is a lightweight
   wrapper for pip to support requirements.txt, Pipenv and Poetry lock files or
   converting them to pip-tools compatible output. Designed for containerized
   Python applications but not limited to them.
+* `pdm <https://github.com/pdm-project/pdm>`_ for a modern Python package management
+  tool supporting `PEP 582 <https://www.python.org/dev/peps/pep-0582/>`_
+  (no virtualenvs: installation of packages in a ``__pypackages__`` folder,
+  *à la* ``node_modules``) and relying on standards such as
+  `PEP 517 <https://www.python.org/dev/peps/pep-0517/>`_ and
+  `PEP 621 <https://www.python.org/dev/peps/pep-0621/>`_.
```
Comment (bhrutledge) [inline source/tutorials/managing-dependencies.rst:176]: We prefer to use the `:pep:` role for readability. I also think the parenthetical could be more succinct, and rely on the PEP for details.

```suggestion
* `PDM <https://github.com/pdm-project/pdm>`_ for a modern Python package management
  tool supporting :pep:`582` (replacing virtual environments with ``__pypackages__``
  directory for package installation) and relying on standards such as :pep:`517` and
  :pep:`621`.
```

File: source/tutorials/managing-dependencies.rst [hunk 2]
```diff
@@ -161,20 +159,26 @@ Other Tools for Application Dependency Management
...
+* `pip-tools <https://github.com/jazzband/pip-tools>`_ to build your own
+  custom workflow from lower level pieces like ``pip-compile`` and ``pip-sync``
```
Comment (bhrutledge) [inline source/tutorials/managing-dependencies.rst:178]:
```suggestion
* `pip-tools <https://github.com/jazzband/pip-tools>`_ for creating a lock file of all
  dependencies from a list of packages directly used in a project, and ensuring that
  only those dependencies are installed.
```

File: source/tutorials/managing-dependencies.rst [hunk 3]
```diff
@@ -161,20 +159,26 @@ Other Tools for Application Dependency Management
...
 
-* `poetry <https://github.com/python-poetry/poetry>`__ for a tool comparable in scope
-  to ``pipenv`` that focuses more directly on use cases where the repository being
-  managed is structured as a Python project with a valid ``pyproject.toml`` file
-  (by contrast, ``pipenv`` explicitly avoids making the assumption that the
-  application being worked on that's depending on components from PyPI will
-  itself support distribution as a ``pip``-installable Python package).
 * `hatch <https://github.com/ofek/hatch>`_ for opinionated coverage of even
   more steps in the project management workflow (such as incrementing versions,
   tagging releases, and creating new skeleton projects from project templates)
```
Comment (bhrutledge) [inline source/tutorials/managing-dependencies.rst:166]:
```suggestion
  more steps in the project management workflow, such as incrementing versions,
  tagging releases, and creating new skeleton projects from project templates.
```

File: source/tutorials/managing-dependencies.rst [hunk 4]
```diff
@@ -161,20 +159,26 @@ Other Tools for Application Dependency Management
...
-* `pip-tools <https://github.com/jazzband/pip-tools>`_ to build your own
-  custom workflow from lower level pieces like ``pip-compile`` and ``pip-sync``
 * `micropipenv <https://github.com/thoth-station/micropipenv>`_ is a lightweight
   wrapper for pip to support requirements.txt, Pipenv and Poetry lock files or
   converting them to pip-tools compatible output. Designed for containerized
   Python applications but not limited to them.
```
Comment (bhrutledge) [inline source/tutorials/managing-dependencies.rst:170]:
```suggestion
* `micropipenv <https://github.com/thoth-station/micropipenv>`_ for a lightweight
  wrapper around pip that supports ``requirements.txt``, Pipenv and Poetry lock files,
  or converting them to pip-tools compatible output. Designed for containerized
  Python applications, but not limited to them.
```

File: source/tutorials/managing-dependencies.rst [hunk 5]
```diff
@@ -161,20 +159,26 @@ Other Tools for Application Dependency Management
...
+  `PEP 517 <https://www.python.org/dev/peps/pep-0517/>`_ and
+  `PEP 621 <https://www.python.org/dev/peps/pep-0621/>`_.
+* `pip-tools <https://github.com/jazzband/pip-tools>`_ to build your own
+  custom workflow from lower level pieces like ``pip-compile`` and ``pip-sync``
+* `poetry <https://github.com/python-poetry/poetry>`__ for a tool comparable in scope
+  to ``pipenv`` that focuses more directly on use cases where the repository being
+  managed is structured as a Python project with a valid ``pyproject.toml`` file
+  (by contrast, ``pipenv`` explicitly avoids making the assumption that the
+  application being worked on that's depending on components from PyPI will
+  itself support distribution as a ``pip``-installable Python package).
```
Comment (bhrutledge) [inline source/tutorials/managing-dependencies.rst:184]:
```suggestion
* `Poetry <https://github.com/python-poetry/poetry>`__ for a tool comparable in scope
  to Pipenv that focuses more directly on use cases where the project being managed is
  structured as a distributable Python package with a valid ``pyproject.toml`` file.
  By contrast, Pipenv explicitly avoids making the assumption that the application
  being worked on will support distribution as a ``pip``-installable Python package.
```

File: source/tutorials/managing-dependencies.rst [hunk 6]
```diff
@@ -21,10 +21,8 @@ applicable to the development and deployment of network services (including
 web applications), but is also very well suited to managing development and
 testing environments for any kind of project.
 
-Developers of Python libraries, or of applications that support distribution
-as Python libraries, should also consider the
-`poetry <https://github.com/python-poetry/poetry>`_ project as an alternative dependency
-management solution.
+Alternative dependency management solutions can be found in a non-exhaustive list
+at the bottom of the page.
```
Comment (bhrutledge) [inline source/tutorials/managing-dependencies.rst:25]: Let's take advantage of [implicit link targets](https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#implicit-targets).

```suggestion
For alternatives, see `Other Tools for Application Dependency Management`_.
```

**Review by bhrutledge — CHANGES_REQUESTED** [review]
Thanks for this, and apologies for the lag on the review. I've got a couple requests re: your edits. I also made some of my own suggestions to make the list more useful; if those feel like too much, I can do them in a separate PR.

File: source/tutorials/managing-dependencies.rst (hunk 4 above)
Comment (pawamoy) [reply source/tutorials/managing-dependencies.rst]: Applied manually (GitHub wouldn't let me).

File: source/tutorials/managing-dependencies.rst (hunk 5 above)
Comment (pawamoy) [reply source/tutorials/managing-dependencies.rst]: Applied manually (GitHub wouldn't let me).

**Comment by pawamoy** [comment]
Hi @bhrutledge, thanks for the review! Nice suggestions, I applied them. Let me know if there's something else to do.

**Review by bhrutledge — APPROVED** [review]
Excellent! Thank you.

**Pull request merged**